src/pages/authentication/Signup.jsx

```diff
@@ -1,74 +1,134 @@
               type="text"
-              id="email"
-              placeholder="johndoe@gmail.com"
-              className="input p-xxs m-xxs m-rl0 bd-rad-sm"
             />
-            <label className="success-msg m-xxs m-rl0">
-              Success! Email entered is correct.
-            </label>
-          </div>
-          <div className="input-wrapper m-xxxs m-rl0">
-            <label htmlFor="password">Password</label>
-            <div className="input-wrapper bd-rad-sm input-w-btn flex flex-row ai-center m-xxs m-rl0">
-              <input
-                type="text"
-                className="p-xxs input"
-                placeholder="password"
-              />
-              <div className="p-xxs gray-text input-btn">
-                <i className="fa-solid fa-eye" />
-              </div>
-            </div>
-            <label className="error-msg">Wrong Password. Try again.</label>
-          </div>
-          <div className="input-wrapper m-xxs m-rl0">
-            <label className="" htmlFor="password">
-              Confirm Password
-            </label>
-            <div className="input-wrapper bd-rad-sm input-w-btn flex flex-row ai-center m-xxs m-rl0">
-              <input
-                type="text"
-                className="p-xxs input"
-                placeholder="confirm password"
-              />
-              <div className="p-xxs gray-text input-btn">
-                <i className="fa-solid fa-eye" />
-              </div>
-            </div>
-            <label className="error-msg">Passwords do not match.</label>
           </div>
+          <TextInput
+            labelText="Email"
+            id="email"
+            name="email"
+            placeholder="johndoe@gmail.com"
+            userData={userData}
+            setUserData={setUserData}
+            type="email"
+          />
+
+          <PasswordInput
+            labelText="Password"
+            id="password"
+            name="password"
+            placeholder="password"
+            userData={userData}
+            setUserData={setUserData}
+          />
+
+          <PasswordInput
+            labelText="Confirm Password"
+            id="confirm-password"
+            name="confirmPassword"
+            placeholder="confirm password"
+            userData={userData}
+            setUserData={setUserData}
+          />
+
           <p className="m-xxs m-rl0 gray-text fs-6">
             By signing up, you agree to our terms and conditions.
           </p>
-          <button type="submit" className="btn btn-primary m-xxs m-rl0">
+          <button
+            type="submit"
+            className="btn btn-primary m-xxs m-rl0"
+            onClick={handleSubmitClick}
+          >
             Sign Up
           </button>
           <p className="m-xxs m-rl0 center-align-text gray-text">
-            Already a User?
-            <a className="primary-color-text" href="/pages/Login/Login.html">
+            Already a User?{" "}
```

Comment:
The `{" "}` could be removed. It comes because of `prettier` by default at some places.